src/app/ReadHandler.cpp

```diff
@@ -578,6 +574,11 @@ const char * ReadHandler::GetStateStr() const
 
 void ReadHandler::MoveToState(const HandlerState aTargetState)
 {
+    if (IsAwaitingReportResponse())
```

Comment:
Should we check that we are moving to a different state?

Reply:
Agree, fixed, thanks